fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -418,6 +426,14 @@ public boolean visit(java.lang.String elementName, int elementIndex, Coding codi
                 p.setValueSystem(coding.getSystem().getValue());
             }
             result.add(p);
+            if (coding.getDisplay() != null && coding.getDisplay().hasValue()) {
+                // Extract as token as normalized string since :text modifier is simple string search
+                p = new TokenParmVal();
+                p.setResourceType(resourceType);
+                p.setName(searchParamCode + SearchConstants.TEXT_MODIFIER_SUFFIX);
+                p.setValueCode(SearchUtil.normalizeForSearch(coding.getDisplay().getValue()));
+                result.add(p);
+            }
```

Comment:
Don't we also want to index the Identifier.type.text field?

Reply:
> Don't we also want to index the Identifier.type.text field?

Interesting thought, but how would you search on that things display text?  I don't think you can layer modifiers like `reference:identifier:text` ...if we supported that I bet we'd be the only ones.

Reply:
I was assuming they were talking about the identifier search parameter - i.e. `identifier:text=abc`, but reading more closely, I guess the spec just says `Identifier.type.text`, which to your point could be any Identifier field, so I'm ok with not supporting that one.

Reply:
Oh, I was thinking of it in terms of being able to search on the text portion of an `of-type`.  But now I see that it does say `:text` is supposed to apply to `Identifier.type.text`.  Weird.  I'm not really sure when/how that would be used, but maybe we *should* support it :shrug:

Reply:
I think it makes sense to support it for search parameters of type TOKEN where the search parameter field is of type Identifier, but not in other cases (like a Reference.identifier field). I also noticed :text is discussed in the spec here: https://www.hl7.org/fhir/search.html#modifiers, and it does not mention searching on the Identifier field, so I'm good with saying we're not supporting it as well:

`For token: :text (the match does a partial searches on the text portion of a CodeableConcept or the display portion of a Coding), instead of the default search which uses codes. Other defined modifiers are :in, :below, :above and :not-in which are described below.
`

Reply:
Yeah, the part of the spec that Mike just referenced is the part I prefer to follow, and feels the most consistent.

There's another section where it lists:

`:text | The search parameter is processed as a string that searches text associated with the code/value - either CodeableConcept.text, Coding.display, or Identifier.type.text.`

However, there's no examples of it and the other section doesn't mention Identifier.type.text. I'm wonder if it just lists it because `Identifier.type` is itself a CodeableConcept, so if a search parameter was defined for it, then it would make sense for :text to then do matches against Identifier.type.text.

Reply:
I support this interpretation.  However, in this case I think we should open a spec bug so we have something to point to if anyone ever asks about it.

Reply:
I opened https://jira.hl7.org/browse/FHIR-31708 for it.